diff -ur old/arch/arm/mach-omap2/omap4-common.c new/arch/arm/mach-omap2/omap4-common.c
--- old/arch/arm/mach-omap2/omap4-common.c	2011-10-24 09:42:58.000000000 +0200
+++ new/arch/arm/mach-omap2/omap4-common.c	2011-12-22 09:35:04.000000000 +0100
@@ -25,7 +25,7 @@
 #include "omap4-sar-layout.h"
 
 #ifdef CONFIG_CACHE_L2X0
-#define L2X0_POR_OFFSET_VALUE		0x9
+#define L2X0_POR_OFFSET_VALUE		0x3
 void __iomem *l2cache_base;
 #endif
 
@@ -129,6 +129,7 @@
 			(1 << 25));
 #endif
 	if (cpu_is_omap446x() || (omap_rev() >= OMAP4430_REV_ES2_2)) {
+		por_ctrl &= ~0x1f;
 		por_ctrl |= L2X0_POR_OFFSET_VALUE;
 		omap_smc1(0x113, por_ctrl);
 	}
