diff -ur old/drivers/staging/omapdrm/omap_crtc.c new/drivers/staging/omapdrm/omap_crtc.c
--- old/drivers/staging/omapdrm/omap_crtc.c	2011-12-25 21:37:25.000000000 +0100
+++ new/drivers/staging/omapdrm/omap_crtc.c	2011-12-25 19:59:47.000000000 +0100
@@ -164,7 +164,7 @@
 	omap_crtc->info.height = mode->vdisplay;
 	omap_crtc->info.out_width = mode->hdisplay;
 	omap_crtc->info.out_height = mode->vdisplay;
-	omap_crtc->info.color_mode = OMAP_DSS_COLOR_RGB24U;
+	omap_crtc->info.color_mode = OMAP_DSS_COLOR_ARGB32;
 	omap_crtc->info.rotation_type = OMAP_DSS_ROT_DMA;
 	omap_crtc->info.rotation = OMAP_DSS_ROT_0;
 	omap_crtc->info.global_alpha = 0xff;
