diff -ur old/Makefile new/Makefile
--- old/Makefile	2010-08-22 13:16:06.000000000 +0200
+++ new/Makefile	2012-12-30 20:59:36.000000000 +0100
@@ -20,7 +20,7 @@
 	install -m 755 run-calibrate.sh $(DESTDIR)$(PREFIX)/bin/run-calibrate.sh
 	install -m 644 X11/Xdefaults $(DESTDIR)/etc/X11/Xdefaults
 	install -m 755 X11/Xinit $(DESTDIR)/etc/X11/Xinit
-	install -m 755 X11/Xserver $(DESTDIR)/etc/X11/Xserver
+	install -m 755 X11/Xserver.sh $(DESTDIR)/etc/X11/Xserver
 	install -m 755 X11/Xsession $(DESTDIR)/etc/X11/Xsession
 	install -m 755 X11/xserver-common $(DESTDIR)/etc/X11/xserver-common
 	install -d $(DESTDIR)/etc/X11/xmodmap
