diff -ur old/Makefile new/Makefile
--- old/Makefile	2011-10-06 07:49:13.000000000 +0200
+++ new/Makefile	2011-10-06 11:08:56.000000000 +0200
@@ -148,14 +148,10 @@
 #########################################################################
 #########################################################################
 
-ALL = $(obj)x-load.bin.ift $(obj)System.map
+ALL = $(obj)x-load.bin $(obj)System.map
 
-all ift:	$(ALL)
+all:	$(ALL)
 
-$(obj)x-load.bin.ift: $(obj)signGP $(obj)System.map $(obj)x-load.bin
-	$(obj)./signGP $(obj)x-load.bin $(TEXT_BASE) $(CONFIG_HEADER)
-	cp $(obj)x-load.bin.ift $(obj)MLO
- 
 $(obj)x-load.bin:	$(obj)x-load
 		$(OBJCOPY) ${OBJCFLAGS} -O binary $< $@
  
@@ -249,7 +245,7 @@
 		| xargs rm -f
 	rm -f $(OBJS) $(obj)*.bak $(obj)tags $(obj)TAGS
 	rm -fr $(obj)*.*~
-	rm -f $(obj)x-load $(obj)x-load.map $(ALL) $(obj)x-load.bin.ift $(obj)signGP $(obj)MLO
+	rm -f $(obj)x-load $(obj)x-load.map $(ALL) $(obj)signGP $(obj)MLO
 	rm -f $(obj)include/asm/proc $(obj)include/asm/arch
 
 ifeq ($(OBJTREE),$(SRCTREE))
