diff -ur old/sunrpc/rpc/types.h new/sunrpc/rpc/types.h
--- old/sunrpc/rpc/types.h	2010-08-19 22:32:31.000000000 +0200
+++ new/sunrpc/rpc/types.h	2013-01-20 16:03:17.000000000 +0100
@@ -69,6 +69,7 @@
 #include <sys/types.h>
 #endif
 
+#define __u_char_defined
 #ifndef __u_char_defined
 typedef __u_char u_char;
 typedef __u_short u_short;
@@ -79,6 +80,7 @@
 typedef __fsid_t fsid_t;
 # define __u_char_defined
 #endif
+#define __daddr_t_defined
 #ifndef __daddr_t_defined
 typedef __daddr_t daddr_t;
 typedef __caddr_t caddr_t;
diff -ur old/sunrpc/Makefile new/sunrpc/Makefile
--- old/sunrpc/Makefile	2013-03-23 19:40:57.000000000 +0100
+++ new/sunrpc/Makefile	2013-03-23 19:56:09.000000000 +0100
@@ -179,11 +179,11 @@
 	$(+link)
 
 $(addprefix $(objpfx)cross-,$(rpcgen-objs)): $(objpfx)cross-%.o: %.c
-	gcc $< -c -D_RPC_THREAD_SAFE_ -D_CROSS_RPCGEN_ \
+	gcc $< -c -D_RPC_THREAD_SAFE_ -D_CROSS_RPCGEN_ $(BUILD_CFLAGS) \
 		$(OUTPUT_OPTION) $(compile-mkdep-flags)
 
 $(objpfx)cross-rpcgen: $(addprefix $(objpfx)cross-,$(rpcgen-objs))
-	gcc $^ -o $@
+	gcc $^ $(BUILD_LDFLAGS) -lintl -o $@
 
 # This makes sure -DNOT_IN_libc is passed for all these modules.
 cpp-srcs-left := $(rpcgen-objs:.o=.c)
diff -ur old/sunrpc/proto.h new/sunrpc/proto.h
--- old/sunrpc/proto.h	2013-03-23 19:51:39.000000000 +0100
+++ new/sunrpc/proto.h	2013-03-23 19:50:36.000000000 +0100
@@ -58,7 +58,7 @@
 
 /* Rather then defining _GNU_SOURCE before including $build's <string.h>
    we just declare stpcpy here.  */
-extern char *stpcpy (char *, const char *);
+//extern char *stpcpy (char *, const char *);
 
 /* Use $build's i18n support as we can't use $host's.  */
 #define _(X) (gettext (X))
