diff -urp old/posix/Makefile new/posix/Makefile
--- old/posix/Makefile	2013-03-23 17:32:19.000000000 +0100
+++ new/posix/Makefile	2013-11-10 13:03:01.000000000 +0100
@@ -382,7 +382,7 @@ $(objpfx)cross-getconf.h: confstr.c
 cross-getconf-CFLAGS = -O -I$(objpfx)
 
 $(objpfx)cross-getconf: cross-getconf.c $(objpfx)cross-getconf.h
-	$(native-compile)
+	gcc cross-getconf.c -I$(objpfx) -o $(objpfx)cross-getconf
 endif
 
 $(objpfx)getconf.speclist: $(getconf-dep)
