diff -urp old/ports/sysdeps/unix/sysv/linux/arm/eabi/libc-do-syscall.S new/ports/sysdeps/unix/sysv/linux/arm/eabi/libc-do-syscall.S
--- old/ports/sysdeps/unix/sysv/linux/arm/eabi/libc-do-syscall.S	2013-06-15 19:19:07.000000000 +0200
+++ new/ports/sysdeps/unix/sysv/linux/arm/eabi/libc-do-syscall.S	2013-06-15 21:46:27.000000000 +0200
@@ -24,7 +24,7 @@
    unwind information - and while there is a reserved prefix in the
    ARM unwind tables for register to register moves, the actual opcodes
    are not defined.  */
-
+#if defined(__thumb__)
 	.thumb
 	.syntax unified
 	.hidden __libc_do_syscall
@@ -47,3 +47,5 @@ ENTRY (__libc_do_syscall)
 	pop	{r7, pc}
 	.fnend
 END (__libc_do_syscall)
+
+#endif /* __thumb__ */
