diff -ur old/arch/arm/mach-omap2/Makefile new/arch/arm/mach-omap2/Makefile
--- old/arch/arm/mach-omap2/Makefile	2011-12-06 19:48:12.000000000 +0100
+++ new/arch/arm/mach-omap2/Makefile	2011-12-06 20:44:05.000000000 +0100
@@ -27,8 +27,8 @@
 obj-$(CONFIG_ARCH_OMAP4)		+= omap44xx-smc.o omap4-common.o
 
 plus_sec := $(call as-instr,.arch_extension sec,+sec)
-AFLAGS_omap-headsmp.o			:=-Wa,-march=armv7-a$(plus_sec)
-AFLAGS_omap44xx-smc.o			:=-Wa,-march=armv7-a$(plus_sec)
+AFLAGS_omap-headsmp.o			:=-Wa,-march=armv7-a+sec
+AFLAGS_omap44xx-smc.o			:=-Wa,-march=armv7-a+sec
 
 # Functions loaded to SRAM
 obj-$(CONFIG_SOC_OMAP2420)		+= sram242x.o
@@ -71,7 +71,7 @@
 obj-$(CONFIG_OMAP_SMARTREFLEX_CLASS3)	+= smartreflex-class3.o
 
 AFLAGS_sleep24xx.o			:=-Wa,-march=armv6
-AFLAGS_sleep34xx.o			:=-Wa,-march=armv7-a$(plus_sec)
+AFLAGS_sleep34xx.o			:=-Wa,-march=armv7-a+sec
 
 ifeq ($(CONFIG_PM_VERBOSE),y)
 CFLAGS_pm_bus.o				+= -DDEBUG
