diff -ur old/Makefile.am new/Makefile.am
--- old/Makefile.am	2012-05-22 21:20:13.000000000 +0200
+++ new/Makefile.am	2013-01-26 09:56:45.000000000 +0100
@@ -1,4 +1,4 @@
-SUBDIRS = src protocol doc tests cursor
+SUBDIRS = src protocol doc
 
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
diff -ur old/src/Makefile.am new/src/Makefile.am
--- old/src/Makefile.am	2012-06-29 23:04:36.000000000 +0200
+++ new/src/Makefile.am	2013-01-26 10:02:56.000000000 +0100
@@ -1,36 +1,14 @@
-lib_LTLIBRARIES = libwayland-server.la libwayland-client.la
 noinst_LTLIBRARIES = libwayland-util.la
 
 include_HEADERS =				\
 	wayland-util.h				\
-	wayland-server-protocol.h		\
-	wayland-server.h			\
-	wayland-client-protocol.h		\
-	wayland-client.h			\
-	wayland-egl.h				\
 	wayland-version.h
 
 libwayland_util_la_SOURCES =			\
-	connection.c				\
 	wayland-util.c				\
 	wayland-util.h				\
-	wayland-os.c				\
-	wayland-os.h				\
 	wayland-private.h
 
-libwayland_server_la_LIBADD = $(FFI_LIBS) libwayland-util.la -lrt -lm
-libwayland_server_la_SOURCES =			\
-	wayland-protocol.c			\
-	wayland-server.c			\
-	wayland-shm.c				\
-	data-device.c				\
-	event-loop.c
-
-libwayland_client_la_LIBADD = $(FFI_LIBS) libwayland-util.la -lrt -lm
-libwayland_client_la_SOURCES =			\
-	wayland-protocol.c			\
-	wayland-client.c
-
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = wayland-client.pc wayland-server.pc
 
