diff -ur old/ld/ld.texinfo new/ld/ld.texinfo
--- old/ld/ld.texinfo	2014-05-11 17:40:55.000000000 +0200
+++ new/ld/ld.texinfo	2014-05-11 18:29:23.000000000 +0200
@@ -7868,7 +7868,7 @@
 @printindex cp
 
 @tex
-% I think something like @colophon should be in texinfo.  In the
+% I think something like should be in texinfo.  In the
 % meantime:
 \long\def\colophon{\hbox to0pt{}\vfill
 \centerline{The body of this manual is set in}
@@ -7879,7 +7879,7 @@
 \centerline{{\sl\fontname\tensl\/}}
 \centerline{are used for emphasis.}\vfill}
 \page\colophon
-% Blame: doc@cygnus.com, 28mar91.
+% Blame: cygnus.com, 28mar91.
 @end tex
 
 @bye
