diff -urp old/Makefile new/Makefile
--- old/Makefile	2013-06-16 10:12:03.000000000 +0200
+++ new/Makefile	2013-06-16 12:22:38.000000000 +0200
@@ -17,26 +17,13 @@ endif
 TESTS = tests
 
 MTD_BINS = \
-	ftl_format flash_erase nanddump doc_loadbios \
-	ftl_check mkfs.jffs2 flash_lock flash_unlock \
-	flash_otp_info flash_otp_dump mtd_debug flashcp nandwrite nandtest \
-	jffs2dump \
-	nftldump nftl_format docfdisk \
-	rfddump rfdformat \
-	serve_image recv_image \
+	mkfs.jffs2 \
 	sumtool jffs2reader
-UBI_BINS = \
-	ubiupdatevol ubimkvol ubirmvol ubicrc32 ubinfo ubiattach \
-	ubidetach ubinize ubiformat ubirename mtdinfo ubirsvol
 
 BINS = $(MTD_BINS)
-BINS += mkfs.ubifs/mkfs.ubifs
-BINS += $(addprefix ubi-utils/,$(UBI_BINS))
-SCRIPTS = flash_eraseall
 
 TARGETS = $(BINS)
 TARGETS += lib/libmtd.a
-TARGETS += ubi-utils/libubi.a
 
 OBJDEPS = $(BUILDDIR)/include/version.h
 
diff -urp old/compr_lzo.c new/compr_lzo.c
--- old/compr_lzo.c	2013-06-16 10:12:03.000000000 +0200
+++ new/compr_lzo.c	2013-06-16 12:03:56.000000000 +0200
@@ -26,7 +26,9 @@
 #include <string.h>
 
 #ifndef WITHOUT_LZO
+#ifndef __APPLE__
 #include <asm/types.h>
+#endif
 #include <linux/jffs2.h>
 #include <lzo/lzo1x.h>
 #include "compr.h"
diff -urp old/compr_zlib.c new/compr_zlib.c
--- old/compr_zlib.c	2013-06-16 10:12:03.000000000 +0200
+++ new/compr_zlib.c	2013-06-16 12:01:45.000000000 +0200
@@ -39,7 +39,9 @@
 #include <zlib.h>
 #undef crc32
 #include <stdio.h>
+#ifndef __APPLE__
 #include <asm/types.h>
+#endif
 #include <linux/jffs2.h>
 #include "common.h"
 #include "compr.h"
diff -urp old/include/mtd/jffs2-user.h new/include/mtd/jffs2-user.h
--- old/include/mtd/jffs2-user.h	2013-06-16 10:12:03.000000000 +0200
+++ new/include/mtd/jffs2-user.h	2013-06-16 11:11:49.000000000 +0200
@@ -9,8 +9,16 @@
 
 /* This file is blessed for inclusion by userspace */
 #include <linux/jffs2.h>
+#ifdef __APPLE__
+#include <machine/endian.h>
+#define __BYTE_ORDER __DARWIN_BYTE_ORDER
+#define __LITTLE_ENDIAN __DARWIN_LITTLE_ENDIAN
+#define __BIG_ENDIAN __DARWIN_BIG_ENDIAN
+typedef off_t loff_t;
+#else
 #include <endian.h>
 #include <byteswap.h>
+#endif
 
 #undef cpu_to_je16
 #undef cpu_to_je32
diff -urp old/include/mtd/mtd-abi.h new/include/mtd/mtd-abi.h
--- old/include/mtd/mtd-abi.h	2013-06-16 10:12:03.000000000 +0200
+++ new/include/mtd/mtd-abi.h	2013-06-16 10:39:56.000000000 +0200
@@ -20,7 +20,14 @@
 #ifndef __MTD_ABI_H__
 #define __MTD_ABI_H__
 
+#ifdef __APPLE__
+typedef unsigned char      __u8;
+typedef unsigned short     __u16;
+typedef unsigned int       __u32;
+typedef unsigned long long __u64;
+#else
 #include <linux/types.h>
+#endif
 
 struct erase_info_user {
 	__u32 start;
diff -urp old/jffs2reader.c new/jffs2reader.c
--- old/jffs2reader.c	2013-06-16 10:12:03.000000000 +0200
+++ new/jffs2reader.c	2013-06-16 12:15:29.000000000 +0200
@@ -78,6 +78,13 @@ BUGS:
 
 #include "mtd/jffs2-user.h"
 #include "common.h"
+#ifdef __APPLE__
+#define bswap_16(x) \
+     ((unsigned short int) ((((x) >> 8) & 0xff) | (((x) & 0xff) << 8)))
+#define bswap_32(x) \
+     ((((x) & 0xff000000) >> 24) | (((x) & 0x00ff0000) >>  8) |               \
+     (((x) & 0x0000ff00) <<  8) | (((x) & 0x000000ff) << 24))
+#endif
 
 #define SCRATCH_SIZE (5*1024*1024)
 
diff -urp old/lib/libfec.c new/lib/libfec.c
--- old/lib/libfec.c	2013-06-16 10:12:03.000000000 +0200
+++ new/lib/libfec.c	2013-06-16 11:00:34.000000000 +0200
@@ -88,6 +88,10 @@ u_long ticks[10];	/* vars for timekeepin
 #define TOCK(x)
 #endif /* TEST */
 
+#ifdef __APPLE__
+typedef unsigned long u_long;
+#endif
+
 /*
  * You should not need to change anything beyond this point.
  * The first part of the file implements linear algebra in GF.
diff -urp old/lib/libmtd.c new/lib/libmtd.c
--- old/lib/libmtd.c	2013-06-16 10:12:03.000000000 +0200
+++ new/lib/libmtd.c	2013-06-16 10:49:58.000000000 +0200
@@ -39,6 +39,11 @@
 #include "libmtd_int.h"
 #include "common.h"
 
+#ifdef __APPLE__
+typedef long long __kernel_loff_t;
+typedef off_t loff_t;
+#endif
+
 /**
  * mkpath - compose full path from 2 given components.
  * @path: the first component
diff -urp old/lib/libmtd_legacy.c new/lib/libmtd_legacy.c
--- old/lib/libmtd_legacy.c	2013-06-16 10:12:03.000000000 +0200
+++ new/lib/libmtd_legacy.c	2013-06-16 10:50:38.000000000 +0200
@@ -46,6 +46,11 @@
 #define PROC_MTD_MAX_LEN   4096
 #define PROC_MTD_PATT      "mtd%d: %llx %x"
 
+#ifdef __APPLE__
+typedef long long __kernel_loff_t;
+typedef off_t loff_t;
+#endif
+
 /**
  * struct proc_parse_info - /proc/mtd parsing information.
  * @mtd_num: MTD device number
diff -urp old/mkfs.jffs2.c new/mkfs.jffs2.c
--- old/mkfs.jffs2.c	2013-06-16 12:03:36.000000000 +0200
+++ new/mkfs.jffs2.c	2013-06-16 12:50:38.000000000 +0200
@@ -70,7 +70,20 @@
 #include <sys/xattr.h>
 #include <sys/acl.h>
 #endif
+#ifdef __APPLE__
+#include <machine/endian.h>
+#define __BYTE_ORDER __DARWIN_BYTE_ORDER
+#define __LITTLE_ENDIAN __DARWIN_LITTLE_ENDIAN
+#define __BIG_ENDIAN __DARWIN_BIG_ENDIAN
+#define bswap_16(x) \
+     ((unsigned short int) ((((x) >> 8) & 0xff) | (((x) & 0xff) << 8)))
+#define bswap_32(x) \
+     ((((x) & 0xff000000) >> 24) | (((x) & 0x00ff0000) >>  8) |               \
+     (((x) & 0x0000ff00) <<  8) | (((x) & 0x000000ff) << 24))
+#define PATH_MAX 1024
+#else
 #include <byteswap.h>
+#endif
 #include <crc32.h>
 #include <inttypes.h>
 
diff -urp old/rbtree.h new/rbtree.h
--- old/rbtree.h	2013-06-16 10:12:03.000000000 +0200
+++ new/rbtree.h	2013-06-16 11:37:57.000000000 +0200
@@ -94,8 +94,10 @@ static inline struct page * rb_insert_pa
 #ifndef	_LINUX_RBTREE_H
 #define	_LINUX_RBTREE_H
 
+#ifndef __APPLE__
 #include <linux/kernel.h>
 #include <linux/stddef.h>
+#endif
 
 struct rb_node
 {
diff -urp old/summary.h new/summary.h
--- old/summary.h	2013-06-16 10:12:03.000000000 +0200
+++ new/summary.h	2013-06-16 12:13:41.000000000 +0200
@@ -12,7 +12,9 @@
 #ifndef JFFS2_SUMMARY_H
 #define JFFS2_SUMMARY_H
 
+#ifndef __APPLE__
 #include <linux/uio.h>
+#endif
 #include <linux/jffs2.h>
 
 #define DIRTY_SPACE(x) do { typeof(x) _x = (x); \
diff -urp old/sumtool.c new/sumtool.c
--- old/sumtool.c	2013-06-16 10:12:03.000000000 +0200
+++ new/sumtool.c	2013-06-16 12:14:12.000000000 +0200
@@ -40,11 +40,25 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/param.h>
+#ifndef __APPLE__
 #include <asm/types.h>
+#endif
 #include <dirent.h>
 #include <mtd/jffs2-user.h>
+#ifdef __APPLE__
+#include <machine/endian.h>
+#define __BYTE_ORDER __DARWIN_BYTE_ORDER
+#define __LITTLE_ENDIAN __DARWIN_LITTLE_ENDIAN
+#define __BIG_ENDIAN __DARWIN_BIG_ENDIAN
+#define bswap_16(x) \
+     ((unsigned short int) ((((x) >> 8) & 0xff) | (((x) & 0xff) << 8)))
+#define bswap_32(x) \
+     ((((x) & 0xff000000) >> 24) | (((x) & 0x00ff0000) >>  8) |               \
+     (((x) & 0x0000ff00) <<  8) | (((x) & 0x000000ff) << 24))
+#else
 #include <endian.h>
 #include <byteswap.h>
+#endif
 #include <getopt.h>
 #include <crc32.h>
 #include "summary.h"
