diff -ur old/scripts/mod/sumversion.c new/scripts/mod/sumversion.c
--- old/scripts/mod/sumversion.c	2007-02-24 00:52:30.000000000 +0100
+++ new/scripts/mod/sumversion.c	2010-06-18 17:49:46.000000000 +0200
@@ -10,6 +10,10 @@
 #include <string.h>
 #include "modpost.h"
 
+#ifndef PATH_MAX
+#define PATH_MAX 1024
+#endif
+
 /*
  * Stolen form Cryptographic API.
  *
diff -ur old/Makefile new/Makefile
--- old/Makefile	2012-02-09 17:15:53.000000000 +0100
+++ new/Makefile	2012-02-09 17:10:49.000000000 +0100
@@ -245,7 +245,7 @@
 
 HOSTCC       = gcc
 HOSTCXX      = g++
-HOSTCFLAGS   = -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 -fomit-frame-pointer
+HOSTCFLAGS   = -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 -fomit-frame-pointer $(HOST_INC)
 HOSTCXXFLAGS = -O2
 
 # Decide whether to build built-in, modular, or both.
