diff -ur old/arch/arm/mach-omap2/omap4-common.c new/arch/arm/mach-omap2/omap4-common.c
--- old/arch/arm/mach-omap2/omap4-common.c	2011-10-05 16:39:51.000000000 +0200
+++ new/arch/arm/mach-omap2/omap4-common.c	2011-11-07 10:57:30.000000000 +0100
@@ -211,8 +211,8 @@
 	l2cache_base = ioremap(OMAP44XX_L2CACHE_BASE, SZ_4K);
 	BUG_ON(!l2cache_base);
 
-	if (omap_rev() == OMAP4430_REV_ES2_0)
-		omap_smc1(0x109, 0x7e470000);
+	if (omap_rev() >= OMAP4430_REV_ES2_0)
+		omap_smc1(0x109, 0x5e470000);
 
 	/* Enable PL310 L2 Cache controller */
 	omap_smc1(0x102, 0x1);
@@ -221,8 +221,8 @@
 	 * 32KB way size, 16-way associativity,
 	 * parity disabled
 	 */
-	if (omap_rev() == OMAP4430_REV_ES2_0)
-		l2x0_init(l2cache_base, 0x7e470000, 0xc0000fff);
+	if (omap_rev() >= OMAP4430_REV_ES2_0)
+		l2x0_init(l2cache_base, 0x5e470000, 0xc0000fff);
 	else
 		l2x0_init(l2cache_base, 0x0e050000, 0xc0000fff);
 
