diff -urNP old/api/include/Makefile.am new/api/include/Makefile.am
--- old/api/include/Makefile.am	2011-12-04 03:22:51.000000000 +0100
+++ new/api/include/Makefile.am	2012-12-30 09:49:01.000000000 +0100
@@ -80,7 +80,7 @@
 ListMPDrv.h \
 std_linux.h \
 NotifyDrvDefs.h \
-String.h \
+String-syslink.h \
 elf32.h \
 NotifyDrvUsr.h \
 symtab.h \
diff -urNP old/api/include/String-syslink.h new/api/include/String-syslink.h
--- old/api/include/String-syslink.h	1970-01-01 01:00:00.000000000 +0100
+++ new/api/include/String-syslink.h	2011-12-04 03:22:51.000000000 +0100
@@ -0,0 +1,94 @@
+/*
+ *  Syslink-IPC for TI OMAP Processors
+ *
+ *  Copyright (c) 2008-2010, Texas Instruments Incorporated
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *  *  Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *
+ *  *  Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *
+ *  *  Neither the name of Texas Instruments Incorporated nor the names of
+ *     its contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ *  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ *  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ *  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ *  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ *  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ *  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ *  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+ *  OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ *  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+/** ============================================================================
+ *  @file   String.h
+ *
+ *  @brief      Defines for String manipulation library.
+ *
+ *  ============================================================================
+ */
+
+
+#ifndef UTILS_STRING_H_0XEFC8
+#define UTILS_STRING_H_0XEFC8
+
+
+#if defined (__cplusplus)
+extern "C" {
+#endif
+
+
+/* =============================================================================
+ *  APIs
+ * =============================================================================
+ */
+/* Function to find offset of string in the given string */
+Int String_find (String as1, String as2);
+
+/* Function to add a string at the end of another one */
+String String_cat (String s1, String s2);
+
+/* Function to return a pointer to the first occurrence of the character c in the string sp */
+String String_chr (String sp, Int c);
+
+/* Function to compare two string */
+Int String_cmp (String s1, String s2);
+
+/* Function to compare two string only first n characters */
+Int String_ncmp (String s1, String s2, UInt32 n);
+
+/* Function to copy strings */
+String String_cpy (String s1, String s2);
+
+/* Function to copy first n bytes from s2 to s1 */
+String String_ncpy (String s1, String s2, UInt32 n);
+
+/* Function to locate a substring */
+String String_str (String haystack, String needle);
+
+/* Function to returns the number of characters in s */
+Int String_len (String s);
+
+/* Function to get string representation of a hex number */
+Int String_hexToStr (String s, UInt32 hex);
+
+/* Function to calculate hash for a string */
+UInt32 String_hash (String s);
+
+
+#if defined (__cplusplus)
+}
+#endif /* defined (__cplusplus) */
+
+#endif /* STRING_H_0X5B4D */
diff -urNP old/api/include/String.h new/api/include/String.h
--- old/api/include/String.h	2011-12-04 03:22:51.000000000 +0100
+++ new/api/include/String.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,94 +0,0 @@
-/*
- *  Syslink-IPC for TI OMAP Processors
- *
- *  Copyright (c) 2008-2010, Texas Instruments Incorporated
- *  All rights reserved.
- *
- *  Redistribution and use in source and binary forms, with or without
- *  modification, are permitted provided that the following conditions
- *  are met:
- *
- *  *  Redistributions of source code must retain the above copyright
- *     notice, this list of conditions and the following disclaimer.
- *
- *  *  Redistributions in binary form must reproduce the above copyright
- *     notice, this list of conditions and the following disclaimer in the
- *     documentation and/or other materials provided with the distribution.
- *
- *  *  Neither the name of Texas Instruments Incorporated nor the names of
- *     its contributors may be used to endorse or promote products derived
- *     from this software without specific prior written permission.
- *
- *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
- *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
- *  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
- *  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
- *  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
- *  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
- *  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
- *  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
- *  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
- *  OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
- *  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
- */
-/** ============================================================================
- *  @file   String.h
- *
- *  @brief      Defines for String manipulation library.
- *
- *  ============================================================================
- */
-
-
-#ifndef UTILS_STRING_H_0XEFC8
-#define UTILS_STRING_H_0XEFC8
-
-
-#if defined (__cplusplus)
-extern "C" {
-#endif
-
-
-/* =============================================================================
- *  APIs
- * =============================================================================
- */
-/* Function to find offset of string in the given string */
-Int String_find (String as1, String as2);
-
-/* Function to add a string at the end of another one */
-String String_cat (String s1, String s2);
-
-/* Function to return a pointer to the first occurrence of the character c in the string sp */
-String String_chr (String sp, Int c);
-
-/* Function to compare two string */
-Int String_cmp (String s1, String s2);
-
-/* Function to compare two string only first n characters */
-Int String_ncmp (String s1, String s2, UInt32 n);
-
-/* Function to copy strings */
-String String_cpy (String s1, String s2);
-
-/* Function to copy first n bytes from s2 to s1 */
-String String_ncpy (String s1, String s2, UInt32 n);
-
-/* Function to locate a substring */
-String String_str (String haystack, String needle);
-
-/* Function to returns the number of characters in s */
-Int String_len (String s);
-
-/* Function to get string representation of a hex number */
-Int String_hexToStr (String s, UInt32 hex);
-
-/* Function to calculate hash for a string */
-UInt32 String_hash (String s);
-
-
-#if defined (__cplusplus)
-}
-#endif /* defined (__cplusplus) */
-
-#endif /* STRING_H_0X5B4D */
diff -urNP old/api/src/ipc/GateMP.c new/api/src/ipc/GateMP.c
--- old/api/src/ipc/GateMP.c	2011-12-04 03:22:51.000000000 +0100
+++ new/api/src/ipc/GateMP.c	2012-12-30 09:46:26.000000000 +0100
@@ -45,7 +45,7 @@
 /* Utilities & OSAL headers */
 #include <Memory.h>
 #include <Trace.h>
-#include <String.h>
+#include <String-syslink.h>
 #include <IGateProvider.h>
 #include <Gate.h>
 
diff -urNP old/api/src/ipc/HeapBufMP.c new/api/src/ipc/HeapBufMP.c
--- old/api/src/ipc/HeapBufMP.c	2011-12-04 03:22:51.000000000 +0100
+++ new/api/src/ipc/HeapBufMP.c	2012-12-30 09:46:33.000000000 +0100
@@ -72,7 +72,7 @@
 /* Utilities headers */
 #include <Trace.h>
 #include <Memory.h>
-#include <String.h>
+#include <String-syslink.h>
 
 /* Module level headers */
 #include <ti/ipc/HeapBufMP.h>
diff -urNP old/api/src/ipc/HeapMemMP.c new/api/src/ipc/HeapMemMP.c
--- old/api/src/ipc/HeapMemMP.c	2011-12-04 03:22:51.000000000 +0100
+++ new/api/src/ipc/HeapMemMP.c	2012-12-30 09:46:37.000000000 +0100
@@ -70,7 +70,7 @@
 /* Utilities headers */
 #include <Trace.h>
 #include <Memory.h>
-#include <String.h>
+#include <String-syslink.h>
 
 /* Module level headers */
 #include <_GateMP.h>
diff -urNP old/api/src/ipc/ListMP.c new/api/src/ipc/ListMP.c
--- old/api/src/ipc/ListMP.c	2011-12-04 03:22:51.000000000 +0100
+++ new/api/src/ipc/ListMP.c	2012-12-30 09:46:47.000000000 +0100
@@ -85,10 +85,10 @@
 #include <Std.h>
 
 /* Osal And Utils  headers */
-#include <String.h>
+#include <String-syslink.h>
 #include <Trace.h>
 #include <Memory.h>
-#include <String.h>
+#include <String-syslink.h>
 #include <_GateMP.h>
 #include <ti/ipc/GateMP.h>
 #include <_SharedRegion.h>
diff -urNP old/api/src/ipc/MessageQ.c new/api/src/ipc/MessageQ.c
--- old/api/src/ipc/MessageQ.c	2011-12-04 03:22:51.000000000 +0100
+++ new/api/src/ipc/MessageQ.c	2012-12-30 09:46:53.000000000 +0100
@@ -122,7 +122,7 @@
 #include <Std.h>
 
 /* Osal And Utils  headers */
-#include <String.h>
+#include <String-syslink.h>
 #include <List.h>
 #include <Trace.h>
 #include <Memory.h>
diff -urNP old/api/src/ipc/MultiProc.c new/api/src/ipc/MultiProc.c
--- old/api/src/ipc/MultiProc.c	2011-12-04 03:22:51.000000000 +0100
+++ new/api/src/ipc/MultiProc.c	2012-12-30 09:46:59.000000000 +0100
@@ -45,7 +45,7 @@
 
 /* OSAL & Utils headers */
 #include <Trace.h>
-#include <String.h>
+#include <String-syslink.h>
 #include <Memory.h>
 
 /* Module level headers */
diff -urNP old/api/src/ipc/NameServer.c new/api/src/ipc/NameServer.c
--- old/api/src/ipc/NameServer.c	2011-12-04 03:22:51.000000000 +0100
+++ new/api/src/ipc/NameServer.c	2012-12-30 09:47:04.000000000 +0100
@@ -97,7 +97,7 @@
 #include <Gate.h>
 #include <Memory.h>
 #include <Trace.h>
-#include <String.h>
+#include <String-syslink.h>
 
 /* Module level headers */
 #include <ti/ipc/NameServer.h>
diff -urNP old/api/src/ipc/SharedRegion.c new/api/src/ipc/SharedRegion.c
--- old/api/src/ipc/SharedRegion.c	2011-12-04 03:22:51.000000000 +0100
+++ new/api/src/ipc/SharedRegion.c	2012-12-30 09:47:14.000000000 +0100
@@ -81,7 +81,7 @@
 
 /* OSAL & Utils headers */
 #include <Memory.h>
-#include <String.h>
+#include <String-syslink.h>
 #include <Trace.h>
 #include <GateMutex.h>
 #include <Gate.h>
diff -urNP old/api/src/rcm/RcmClient.c new/api/src/rcm/RcmClient.c
--- old/api/src/rcm/RcmClient.c	2011-12-04 03:22:51.000000000 +0100
+++ new/api/src/rcm/RcmClient.c	2012-12-30 09:47:19.000000000 +0100
@@ -53,7 +53,7 @@
 #include <OsalSemaphore.h>
 #include <Memory.h>
 #include <List.h>
-#include <String.h>
+#include <String-syslink.h>
 
 /* Module level headers */
 #include <RcmTypes.h>
diff -urNP old/api/src/rcm/RcmServer.c new/api/src/rcm/RcmServer.c
--- old/api/src/rcm/RcmServer.c	2011-12-04 03:22:51.000000000 +0100
+++ new/api/src/rcm/RcmServer.c	2012-12-30 09:47:23.000000000 +0100
@@ -56,7 +56,7 @@
 #include <OsalSemaphore.h>
 #include <Memory.h>
 #include <List.h>
-#include <String.h>
+#include <String-syslink.h>
 
 /* Module level headers */
 #include <RcmTypes.h>
diff -urNP old/api/src/sysmgr/Ipc.c new/api/src/sysmgr/Ipc.c
--- old/api/src/sysmgr/Ipc.c	2011-12-04 03:22:51.000000000 +0100
+++ new/api/src/sysmgr/Ipc.c	2012-12-30 09:47:28.000000000 +0100
@@ -47,7 +47,7 @@
 #include <Gate.h>
 #include <Memory.h>
 #include <Trace.h>
-#include <String.h>
+#include <String-syslink.h>
 
 /* Module headers */
 #include <ti/ipc/MultiProc.h>
diff -urNP old/api/src/utils/String.c new/api/src/utils/String.c
--- old/api/src/utils/String.c	2011-12-04 03:22:51.000000000 +0100
+++ new/api/src/utils/String.c	2012-12-30 09:47:33.000000000 +0100
@@ -44,7 +44,7 @@
 #include <Std.h>
 
 /* Module level headers */
-#include <String.h>
+#include <String-syslink.h>
 #include <Trace.h>
 
 
diff -urNP old/daemons/syslink/SyslinkDaemon.c new/daemons/syslink/SyslinkDaemon.c
--- old/daemons/syslink/SyslinkDaemon.c	2011-12-04 03:22:51.000000000 +0100
+++ new/daemons/syslink/SyslinkDaemon.c	2012-12-30 09:49:37.000000000 +0100
@@ -58,7 +58,7 @@
 /* OSAL & Utils headers */
 #include <OsalPrint.h>
 #include <Memory.h>
-#include <String.h>
+#include <String-syslink.h>
 
 /* IPC headers */
 #include <IpcUsr.h>
