diff -urp old/configure.ac new/configure.ac
--- old/configure.ac	2013-05-18 16:54:35.000000000 +0200
+++ new/configure.ac	2013-11-10 16:48:35.000000000 +0100
@@ -50,21 +50,9 @@ dnl Check for libdrm
 PKG_CHECK_MODULES(DRM, libdrm libdrm_omap)
 
 dnl Check for X11/libdri2
-PKG_CHECK_MODULES(X11, x11 dri2, [HAVE_X11=yes], [HAVE_X11=no])
-if test "x$HAVE_X11" = "xyes"; then
-  AC_DEFINE(HAVE_X11, 1, [Have X11 support])
-else
-  AC_MSG_WARN([Building without X11 support])
-fi
 AM_CONDITIONAL(ENABLE_X11, [test "x$HAVE_X11" = xyes])
 
 dnl Check for wayland
-PKG_CHECK_MODULES(WAYLAND, wayland-client >= 0.95.0, [HAVE_WAYLAND=yes], [HAVE_WAYLAND=no])
-if test "x$HAVE_WAYLAND" = "xyes"; then
-  AC_DEFINE(HAVE_WAYLAND, 1, [Have Wayland support])
-else
-  AC_MSG_WARN([Building without Wayland support])
-fi
 AM_CONDITIONAL(ENABLE_WAYLAND, [test "x$HAVE_WAYLAND" = xyes])
 
 dnl *** checks for header files ***
