diff -urp old/wayland-drm-protocol.c new/wayland-drm-protocol.c
--- old/wayland-drm-protocol.c	2013-05-18 16:54:35.000000000 +0200
+++ new/wayland-drm-protocol.c	2013-11-10 14:15:00.000000000 +0100
@@ -66,7 +66,7 @@ static const struct wl_message wl_drm_ev
 
 WL_EXPORT const struct wl_interface wl_drm_interface = {
 	"wl_drm", 1,
-	ARRAY_LENGTH(wl_drm_requests), wl_drm_requests,
-	ARRAY_LENGTH(wl_drm_events), wl_drm_events,
+	sizeof(wl_drm_requests)/sizeof(struct wl_message), wl_drm_requests,
+	sizeof(wl_drm_events)/sizeof(struct wl_message), wl_drm_events,
 };
 
