diff -urp old/Makefile new/Makefile
--- old/Makefile	2013-01-05 20:17:05.000000000 +0100
+++ new/Makefile	2013-01-05 20:26:47.000000000 +0100
@@ -245,7 +245,7 @@ CONFIG_SHELL := $(shell if [ -x "$$BASH"
 
 HOSTCC       = gcc
 HOSTCXX      = g++
-HOSTCFLAGS   = -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 -fomit-frame-pointer
+HOSTCFLAGS   = -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 -fomit-frame-pointer $(HOST_INC)
 HOSTCXXFLAGS = -O2
 
 # Decide whether to build built-in, modular, or both.
diff -urp old/scripts/mod/sumversion.c new/scripts/mod/sumversion.c
--- old/scripts/mod/sumversion.c	2013-01-05 20:17:13.000000000 +0100
+++ new/scripts/mod/sumversion.c	2013-01-05 20:28:53.000000000 +0100
@@ -10,6 +10,10 @@
 #include <limits.h>
 #include "modpost.h"
 
+#ifndef PATH_MAX
+#define PATH_MAX 1024
+#endif
+
 /*
  * Stolen form Cryptographic API.
  *
diff -urN old/scripts/genksyms/parse.y new/scripts/genksyms/parse.y
--- old/scripts/genksyms/parse.y	2008-07-24 02:41:35.000000000 +0200
+++ new/scripts/genksyms/parse.y	2009-12-19 15:51:53.000000000 +0100
@@ -24,7 +24,11 @@
 %{
 
 #include <assert.h>
+#ifdef __APPLE__
+#include <malloc/malloc.h>
+#else
 #include <malloc.h>
+#endif
 #include "genksyms.h"
 
 static int is_typedef;
--- old/scripts/genksyms/parse.c_shipped	2007-02-24 00:52:30.000000000 +0100
+++ new/scripts/genksyms/parse.c_shipped	2010-06-19 05:56:06.000000000 +0200
@@ -144,7 +144,11 @@
 
 
 #include <assert.h>
+#ifdef __APPLE__
+#include <malloc/malloc.h>
+#else
 #include <malloc.h>
+#endif
 #include "genksyms.h"
 
 static int is_typedef;
