diff -ur old/configure new/configure
--- old/configure	2011-12-20 23:21:47.000000000 +0100
+++ new/configure	2011-12-25 08:38:49.000000000 +0100
@@ -8174,8 +8174,8 @@
 #define OUTBURST 512
 
 /* Enable fast OSD/SUB renderer (looks ugly, but uses less CPU power) */
-#undef FAST_OSD
-#undef FAST_OSD_TABLE
+#define FAST_OSD
+#define FAST_OSD_TABLE
 
 /* Define this to enable MPEG-1/2 image postprocessing in libmpeg2 */
 #define MPEG12_POSTPROC 1
