diff -ur old/fmt-conversion.c new/fmt-conversion.c
--- old/fmt-conversion.c	2011-12-25 17:10:28.000000000 +0100
+++ new/fmt-conversion.c	2011-12-25 17:10:51.000000000 +0100
@@ -57,7 +57,7 @@
     {IMGFMT_RGB8,    PIX_FMT_BGR8},
     {IMGFMT_RGB4,    PIX_FMT_BGR4},
     {IMGFMT_BGR8,    PIX_FMT_PAL8},
-    {IMGFMT_BGR32,   PIX_FMT_0RGB32},
+    {IMGFMT_BGR32,   PIX_FMT_RGB32},
 #if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(51, 20, 1)
     {IMGFMT_GBR24P,  PIX_FMT_GBR24P},
 #endif
