From ef866c7293e699b119ae61738e221857a36a5362 Mon Sep 17 00:00:00 2001
From: Jesse Barnes <jbarnes@virtuousgeek.org>
Date: Sat, 2 Feb 2013 11:10:24 +0100
Subject: [PATCH 5/7] intel: add more VLV PCI IDs

---
 intel/intel_chipset.h |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/intel/intel_chipset.h b/intel/intel_chipset.h
index a2eb894..ebec2f8 100644
--- a/intel/intel_chipset.h
+++ b/intel/intel_chipset.h
@@ -84,6 +84,9 @@
 #define PCI_CHIP_HASWELL_CRW_S_GT2_PLUS 0x0D3A
 
 #define PCI_CHIP_VALLEYVIEW_PO		0x0f30 /* power on board */
+#define PCI_CHIP_VALLEYVIEW_1		0x0f31
+#define PCI_CHIP_VALLEYVIEW_2		0x0f32
+#define PCI_CHIP_VALLEYVIEW_3		0x0f33
 
 #define IS_830(dev) (dev == 0x3577)
 #define IS_845(dev) (dev == 0x2562)
@@ -124,7 +127,10 @@
 
 #define IS_I965GM(dev) (dev == 0x2A02)
 
-#define IS_VALLEYVIEW(dev) (dev == 0xf30)
+#define IS_VALLEYVIEW(dev) (((dev) == PCI_CHIP_VALLEYVIEW_PO) ||	\
+			    ((dev) == PCI_CHIP_VALLEYVIEW_1) ||		\
+			    ((dev) == PCI_CHIP_VALLEYVIEW_2) ||		\
+			    ((dev) == PCI_CHIP_VALLEYVIEW_3))
 
 #define IS_GEN4(dev) (dev == 0x2972 ||	\
 		      dev == 0x2982 ||	\
-- 
1.7.9.5

