From: Xavier Boudet <x-boudet@ti.com>
Date: Fri, 14 Sep 2012 15:53:08 +0200
Subject: Update to libudev.pc name
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Xavier Boudet <x-boudet@ti.com>
Signed-off-by: Vincent Stehlé <v-stehle@ti.com>
---
 Makefile.am  |    4 ++--
 configure.ac |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 740ea6e..b14a081 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -6,10 +6,10 @@ libgbm_la_SOURCES =	\
 	common.c
 
 libgbm_la_LIBADD =	\
-	@UDEV_LIBS@
+	@LIBUDEV_LIBS@
 
 libgbm_la_CFLAGS =	\
-	@UDEV_CFLAGS@	\
+	@LIBUDEV_CFLAGS@	\
 	-D_OS_UNIX=1	\
 	-DMODULEDIR='"$(exec_prefix)/lib/gbm"'
 
diff --git a/configure.ac b/configure.ac
index 2739092..a6ddfd9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -35,8 +35,8 @@ AM_MAINTAINER_MODE
 AC_PROG_LIBTOOL
 
 # Obtain compiler/linker options for dependencies
-PKG_CHECK_MODULES(UDEV, udev)
-PKG_CHECK_MODULES(DRM, libdrm)
+PKG_CHECK_MODULES([LIBUDEV], [libudev])
+PKG_CHECK_MODULES([DRM], [libdrm])
 
 AC_CONFIG_FILES([Makefile gbm.pc])
 AC_OUTPUT
-- 
