diff -ur old/tools/os_support.c new/tools/os_support.c
--- old/tools/os_support.c	2011-10-05 19:56:01.000000000 +0200
+++ new/tools/os_support.c	2011-10-06 10:02:54.000000000 +0200
@@ -24,5 +24,5 @@
 #include "mingw_support.c"
 #endif
 #ifdef __APPLE__
-#include "getline.c"
+//#include "getline.c"
 #endif
diff -ur old/tools/os_support.h new/tools/os_support.h
--- old/tools/os_support.h	2011-10-06 10:04:43.000000000 +0200
+++ new/tools/os_support.h	2011-10-06 10:04:30.000000000 +0200
@@ -29,7 +29,7 @@
 #endif
 
 #ifdef __APPLE__
-#include "getline.h"
+//#include "getline.h"
 #endif
 
 #endif /* __OS_SUPPORT_H_ */
