diff -ur old/gcc/Makefile.in new/gcc/Makefile.in
--- old/gcc/Makefile.in	2011-10-03 09:58:33.000000000 +0200
+++ new/gcc/Makefile.in	2011-10-03 10:00:11.000000000 +0200
@@ -769,7 +769,6 @@
 BUILD_LINKERFLAGS = $(BUILD_CFLAGS)
 
 # Native linker and preprocessor flags.  For x-fragment overrides.
-BUILD_LDFLAGS=@BUILD_LDFLAGS@
 BUILD_CPPFLAGS=$(INCLUDES) @BUILD_CPPFLAGS@ $(X_CPPFLAGS)
 
 # Actual name to use when installing a native compiler.
